### PR TITLE
Using unique string for cipher generation

### PR DESCRIPTION
**Pull request by jyotisingh**
re-generation of cipher and re-encryption of secure values in config when unsafe cipher is available

### CODE REVIEW

File: server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                } catch (IOException e1) {
+                    LOGGER.error("Could not replace the cipher file [{}] with original one [{}], please do so manually. Error: [{}]", cipherFile.getAbsolutePath(), originalCipherFile.getAbsolutePath(), e.getMessage(), e);
+                    bomb(e1);
+                }
+            }
+        }
+    }
+
+    private String reEncryptUsingNewKey(byte[] oldCipher, byte[] newCipher, String encryptedValue) throws InvalidCipherTextException {
+        GoCipher cipher = new GoCipher();
```
Comment (arvindsv) [inline server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java]: Can make this a field, if needed. Ok to leave it in too. I thought that every call to `CipherProvider#primeKeyCache` reads the cipher file. But, it is `private static volatile`. So, it doesn't.

File: server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            byte[] oldCipher = FileUtils.readFileToByteArray(originalCipherFile);
+            new CipherProvider(systemEnvironment).resetCipher();
+
+            byte[] newCipher = FileUtils.readFileToByteArray(cipherFile);
+
+            if (new String(newCipher).equals(new String(oldCipher))) {
+                LOGGER.warn("Unable to generate a new safe cipher. Your cipher is unsafe.");
+                FileUtils.deleteQuietly(originalCipherFile);
+                FileUtils.deleteQuietly(originalConfigFile);
+                return;
```
Comment (ketan) [inline server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java]: We return without restoring the old cipher and config xml?

Comment (arvindsv) [inline server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java]: They were never moved, right?

Comment (jyotisingh) [inline server/src/com/thoughtworks/go/config/ConfigCipherUpdater.java]: point noted, renamed variable

**Pull request merged**